recipes/libcurl/all/conanfile.py

```diff
@@ -249,7 +249,7 @@ def _get_configure_command_args(self):
             params.append("--without-wolfssl")
 
         if self.options.with_libssh2:
-            params.append("--with-libssh2={}".format(tools.unix_path(self.deps_cpp_info["libssh2"].lib_paths[0])))
+            params.append("--with-libssh2={}".format(tools.unix_path(self.deps_cpp_info["libssh2"].rootpath)))
```

Comment:
Same than zlib: https://github.com/curl/curl/blob/f3294d9d86e6a7915a967efff2842089b8b0d071/configure.ac#L2800

Reply:
not everything has to be done in this PR, SSE4 did an awesome job with this improvement,  adding more load on him is maybe not fair and nice. And there can also be space for future improvements :-)